components/tv/TVActionSheetProvider.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {
+  createContext,
+  type ReactNode,
+  useCallback,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+import {
+  type TVActionSheetProps,
+  type TVActionSheetRef,
+  TVModalActionSheet,
+} from "./TVModalActionSheet";
+
+interface TVActionSheetContextValue {
+  showActionSheet: (props: Omit<TVActionSheetProps, "visible">) => void;
+  hideActionSheet: () => void;
+}
+
+const TVActionSheetContext = createContext<TVActionSheetContextValue | null>(
+  null,
+);
+
+interface TVActionSheetProviderProps {
+  children: ReactNode;
+}
+
+export const TVActionSheetProvider: React.FC<TVActionSheetProviderProps> = ({
```

Comment:
i kinda want to is to have a single
sheet interface for both mobile and TV

@fredrikburmester i think you had some thoughts on it aswell?

Reply:
What do you think, if the `TVActionSheetProvider`  and `TVActionSheet` , (will be renamed to `ActionSheetProvider` and `ActionSheet` ) will replace the expo native action sheet completely? Right now I am unable to build a mobile version to test it , but it should work ootb

For now we will remain the [BottomSheet´s](https://github.com/gorhom/react-native-bottom-sheet) , because from what i´ve seen the BottomSheets only provide information´s and not any actions.